acceptance/builder_test.go

```diff
@@ -0,0 +1,367 @@
+				})
+			})
+
+			// .../cmd/lifecycle/builder.go#Exec
+			when("invalid builpack api group toml file", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/invalid_buildpack_api_group.toml",
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to : parse buildpack API"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#readData
+		when("parse plan.toml", func() {
+			when("no default plan.toml file in default location", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to parse detect plan"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			//TODO: check some output file for this case not except any error message
+			when("empty parse plan.toml file", func() {
+				it("success", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/empty_plan.toml",
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					_, err := command.CombinedOutput()
+					//print(string(output), err)
+					h.AssertNil(t, err)
+					//expected := "failed to read buildpack order file"
+					//h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			when("invalid parse plan.toml file", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/wrong_plan.toml",
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to parse detect plan: Near line"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+		})
+	})
+
+	when("PlaceHolders need to replace with defaults", func() {
+
+		// .../cmd/lifecycle/builder.go#Args
+		when("groupPath is equals to PlaceHolder groupPath", func() {
+			it("will replace placeholder with default groupPath", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=<layers>/group.toml",
```

Comment:
This test is a bit weird to me, because even though it does demonstrate that the code is correct, I don't think anyone would provide `<layers>/group.toml` as the group path in "real life". 

What if instead, we set the `CNB_LAYERS_DIR` to be something other than `/layers`, and provide a group.toml in that directory. So there would be a `/some-other-layers/group.toml` in the fixture, and we could assert that that group.toml was used?

Reply:
Wow, this test is sounds great for me let me add.

Reply:
I created "acceptance/testdata/builder/container/layers/different_layer_dir_from_env" and i used this folder for tests. Due to permission errors i have to use this directory. Please see this line in dockerfile. https://github.com/buildpacks/lifecycle/pull/648/files#diff-9edd37c29964124e391169e05cb6d5917c38c1dab1922047eaf5fefa2d77dbcaR13

Reply:
That works!